### ISSUE TITLE
Default Http Client for Google_Client has wrong default setting

### ISSUE THREAD

**Issue by sylvain-msl** [googleapis/google-api-php-client#1964]
#### Description

Google Api Client `Google_Client` works with Guzzle Http starting from version 5.0 to version 7.0. Version 7 introduced a deprecation for `exceptions` key in config options. It is currently used in `createDefaultHttpClient` https://github.com/googleapis/google-api-php-client/blob/d0dcc7054faad3cdda5844e15103a59984969744/src/Google/Client.php#L1151

For version 7 and up `http_errors` should be used instead.

This can lead to unexpected behavior, for instance when used in pair with the Google `OAuthClient` that is not catching Guzzle `RequestExceptions`

#### Environment details

  - OS: Linux Debian
  - PHP version: 7.2.28
  - Package name and version: google/apiclient v2.7.2 and guzzlehttp/guzzle 7.0

#### Steps to reproduce

  1. Create a new `Google_Client`
  1. Set `refresh_token` with a non working token
  1. Catch a Guzzle `RequestException` instead of `Google_Service_Exception`

#### Code example

```php
use GuzzleHttp\Exception\RequestException;
use Google_Service_Exception;

try {
    $client = new Google_Client();
    $client->setClientId('an_id');
    $client->setClientSecret('a_secret');
    $client->refreshToken('a_wrong_refresh_token');
} catch (Google_Service_Exception $exception) {
    echo 'google exception';
} catch (RequestException $exception) {
    echo 'not google exception';
}
```

#### Proposed Quick Fix

After https://github.com/googleapis/google-api-php-client/blob/d0dcc7054faad3cdda5844e15103a59984969744/src/Google/Client.php#L1151
add
```php
if (7 === $guzzleVersion && isset($options['exceptions']) {
    $options['http_errors'] = $options['exceptions'];
    unset($options['exceptions']);
}
```

**Comment by jdpedrie** [comment]
Hi @sylvain-msl, thanks for bringing this to our attention. The proposed solution seems good to me. Would you like to open a pull request?

**Comment by sylvain-msl** [comment]
@jdpedrie Sure. will do.

**Comment by bshaffer** [comment]
Fixed in #1966
